src/config.ml

```diff
@@ -2278,7 +2278,11 @@ let client_generate_connect_options t =
         | B (Link_mtu, _) -> true
         | B (Pull, _) -> true
         | B (Tls_mode, `Client) -> true
-        | B (Auth, _) -> true
+        | B (Auth, _) ->
```

Comment:
We should send `auth` also if we use non-AEAD cipher. If the assumption is we always use AEAD ciphers we should document this assumption IMO.

Reply:
we should document that invariant, indeed. From my perspective, the only use of AES-CBC is in the static mode.

Reply:
I added documentation in the README as well as in the code. While writing this I realized it should perhaps be:

```suggestion
        | B (Auth, _) -> Conf_map.mem Tls_auth t
```

As the `--auth` option is used for 1) static clients, 2) tls-auth; and for static clients we don't send connect options.

Reply:
This looks fine, but isn't `Secret` the same? I mean, wasn't `Secret` an alias for `Tls_auth`?

Reply:
ah no, nevermind.